packages/compass-query-bar/src/stores/ai-query-reducer.ts

```diff
@@ -243,6 +255,34 @@ export const runAIQuery = (
       return;
     }
 
+    // The query endpoint may return the aggregation property in addition to filter, project, etc..
+    // It happens when the AI model couldn't generate a query and tried to fulfill a task with the aggregation.
+    const aggregation = Object.entries(query ?? {}).find(([key, value]) => {
+      return (
+        key === 'aggregation' &&
+        typeof value === 'object' &&
+        value !== null &&
+        typeof (value as any).pipeline === 'string'
+      );
+    });
+
+    if (aggregation) {
```

Comment:
Should we check that there is no query returned with aggergation / move this `if (aggregation)` block under the check that comes afterwards that determines that query came back empty? When testing backend locally, I saw sometimes that returned response included both query and aggergation sometimes, something like `{query: { find: '{ _id: 1 }', aggregation: { pipeline: '[{ $match: { _id: 1 } }]' } }`, which is a valid query and we probably don't need to switch in this case?